src/supervised/supervised.jl

```diff
@@ -496,16 +402,15 @@ for (FUN, DESC, EXAMPLE) in (
                     numbers::AbstractArray{T,N},
                     ::AggMode.Mean) where {T,N}
                 nrm = 1 / length(numbers)
-                S = typeof(($FUN)(loss, one(T)) * one(nrm))
-                mapreduce(x -> loss(x) * nrm, +, numbers)::S
+                mapreduce(x -> ($FUN)(loss, x) * nrm, +, numbers)
             end
 
             # Compute the sum (returns a Number)
             function ($FUN)(
                     loss::$KIND,
                     numbers::AbstractArray{T,N},
                     ::AggMode.Sum) where {T,N}
-                mapreduce(loss, +, numbers)
+                mapreduce(x -> ($FUN)(loss, x), +, numbers)
```

Comment:
why not
```suggestion
                sum(x -> ($FUN)(loss, x), numbers)
```

Reply:
Are there performance benefits @oxinabox ? I chose the same mapreduce of the ``AggMode.Mean`` method above for consistency, but would be happy to use `sum` instead if you prefer.